react/features/visitors/middleware.ts

```diff
@@ -359,16 +359,42 @@ function _subscribeVisitorsList(getState: IStore['getState'], dispatch: IStore['
 
     logger.debug('Starting visitors list subscription');
 
+    let initialListReceived = false;
+    let timeoutId: ReturnType<typeof setTimeout>;
+
+    // Set up 10-second timeout for initial list
+    const setupTimeout = () => {
+        timeoutId = setTimeout(() => {
+            if (!initialListReceived) {
+                logger.warn('Initial visitors list not received within 10 seconds, reconnecting...');
+
+                // Disconnect and reconnect
+                VisitorsListWebsocketClient.getInstance().disconnect()
+                    .then(() => {
+                        // Retry the subscription after a brief delay
+                        setTimeout(() => {
+                            _subscribeVisitorsList(getState, dispatch);
+                        }, 1000);
+                    });
+            }
+        }, 10000); // 10 seconds
+    };
+
+    setupTimeout();
```

Comment:
shouldn't we do this on the connect handler not here? I think right now we will end up with 2 timers.